spec/controllers/accounts_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+describe AccountsController, type: :controller do
+  describe "GET #index" do
+    let! (:accounts) {FactoryGirl.create_list(:account, 20)}
+    before do
+        get :index
+    end
+    it "returns http success" do
+        expect(response).to have_http_status(:success)
+    end
+    it "JSON body response contains expected account attributes" do
+      json_response = JSON.parse(response.body)
+      json_response.each do |account|
+        expect(account.keys).to include("id", "name")
+      end
+    end
+    it "get ten accounts" do
+      json_response = JSON.parse(response.body)
+      expect(json_response.length).to be(10)
+    end  
+  end 
+  describe "GET #show" do
+    let! (:accounts) {FactoryGirl.create_list(:account, 20)}
+    before do
+        get :show, params: {:id => accounts[0].id}
+    end
+    it "returns http success" do
+        expect(response).to have_http_status(:success)
+    end
+    it "JSON body response contains expected account attributes" do
+      json_response = JSON.parse(response.body)
+      expect(json_response.keys).to include("id", "name")
+    end
+    it "JSON body response contains match attributes" do
+      json_response = JSON.parse(response.body)
+      expect(json_response.keys).to include("id", "name")
+      expect(json_response["name"]).to match(accounts[0][:name])
+    end
+  end 
+  describe "POST #create" do
+    let (:account) {FactoryGirl.attributes_for :account}
+    before do
+      post :create, params: {:account => account}
+    end
+    it "returns http success" do
+        expect(response).to have_http_status(:success)
+    end 
+    it "JSON body response match account attributes" do
+      json_response = JSON.parse(response.body)
+      expect(json_response.keys).to include("id", "name")
```

Comment:
You could use a schema matcher to ensure the json structure is correct.
https://github.com/ruby-json-schema/json-schema
(Not necessary for this training though)